substrate-node/pallets/pallet-tft-price/src/lib.rs

```diff
@@ -59,119 +73,160 @@ pub mod crypto {
         AccountUnauthorizedToSetPrice,
     }
-}
 
-decl_module! {
-    pub struct Module<T: Config> for enum Call where origin: T::Origin {
-        type Error = Error<T>;
-
-        fn deposit_event() = default;
-
-        #[weight = 100_000_000 + T::DbWeight::get().writes(3)]
-        pub fn set_prices(origin, price: u32, block_number: T::BlockNumber) -> DispatchResultWithPostInfo {
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::storage]
+    #[pallet::getter(fn tft_price)]
+    pub type TftPrice<T> = StorageValue<_, u32, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn last_block_set)]
+    pub type LastBlockSet<T: Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn average_tft_price)]
+    pub type AverageTftPrice<T> = StorageValue<_, u32, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn get_value)]
+    pub type TftPriceHistory<T> = StorageMap<_, Blake2_128Concat, BufferIndex, u32, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn range)]
+    pub type BufferRange<T> = StorageValue<_, (BufferIndex, BufferIndex), ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn allowed_origin)]
+    pub type AllowedOrigin<T: Config> = StorageValue<_, T::AccountId, OptionQuery>;
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::weight(0)]
+        pub fn set_prices(
+            origin: OriginFor<T>,
+            price: u32,
+            block_number: T::BlockNumber,
+        ) -> DispatchResultWithPostInfo {
             let address = ensure_signed(origin)?;
             if let Some(allowed_origin) = AllowedOrigin::<T>::get() {
-                ensure!(allowed_origin == address, Error::<T>::AccountUnauthorizedToSetPrice);
+                ensure!(
+                    allowed_origin == address,
+                    Error::<T>::AccountUnauthorizedToSetPrice
+                );
                 Self::calculate_and_set_price(price, block_number)?;
             }
             Ok(().into())
         }
 
-        #[weight = 100_000_000 + T::DbWeight::get().writes(3)]
-        pub fn set_allowed_origin(origin, target: T::AccountId) {
+        #[pallet::weight(0)]
+        pub fn set_allowed_origin(
+            origin: OriginFor<T>,
+            target: T::AccountId,
+        ) -> DispatchResultWithPostInfo {
             T::RestrictedOrigin::ensure_origin(origin)?;
             AllowedOrigin::<T>::set(Some(target));
+            Ok(().into())
         }
+    }
 
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
         fn offchain_worker(block_number: T::BlockNumber) {
             match Self::offchain_signed_tx(block_number) {
                 Ok(_) => log::info!("offchain worker done."),
-                Err(err) => log::info!("err: {:?}", err)
+                Err(err) => log::info!("err: {:?}", err),
```

Comment:
I think it would be nice to print it using log::error instead. Then you can drop the prefix "err:"

Reply:
Done
Also did it for other pallets